pkg/testutil/testregistry/testregistry_linux.go

```diff
@@ -468,3 +478,17 @@ func NewWithBasicAuth(base *testutil.Base, user, pass string, port int, tls bool
 	}
 	return NewRegistry(base, ca, port, auth, nil)
 }
+
+func SafeRandomString(n int) string {
+	b := make([]byte, n)
+	l, err := rand.Read(b)
+	if err != nil {
+		panic(err)
+	}
+	if l != n {
+		panic(fmt.Errorf("expected %d bytes, got %d bytes", n, l))
+	}
+	// XXX WARNING there is something in the registry (or more likely in the way we generate htpasswd files)
+	// that is broken and does not resist truly random strings
```

Comment:
what does this mean? what does this break?

Reply:
Moved from another file.

Last I checked, either https://github.com/containerd/nerdctl/blob/3c7de2a7cadf0f15d2e2dbadac9475276824d74f/pkg/testutil/testregistry/testregistry_linux.go#L233-L236 was not happy with random strings, or the registry in basic auth mode was not - the end result being that authentication is failing (which is why we base64 encode the string).